09_json/runnerdave/pkg/puppy/types.go

```diff
@@ -0,0 +1,24 @@
+package puppy
+
+import "encoding/json"
+
+// Puppy represents an item in the puppy store
+type Puppy struct {
+	Breed  string  `json:"breed"`
+	Colour string  `json:"color"`
```

Comment:
Just a passing comment, you can add an `omitempty` tag when you want the marshaller to ignore empty fields.
eg:
```
Colour string `json:"colour,omitempty"`
```
Will cause the following json output
```
in := Puppy{ID: 1, Breed: "Poodle", Value: 10}
out := `{"id": 1, "breed": "Poodle", "value": 10}` // No "colour" field
```

Reply:
that is pretty cool, did it for color only, to highlight the difference in the examples